sortinghat/core/api.py

```diff
@@ -771,59 +813,60 @@ def _merge_enrollments(trxl, from_uid, to_uid):
 
         return to_uid
 
-    def _merge_profiles(from_uid, to_uid):
-        """Merge the profiles from two `UniqueIdentity` objects"""
-        if from_uid.profile.is_bot or to_uid.profile.is_bot:
-            to_uid.profile.is_bot = True
-        if not to_uid.profile.name:
-            to_uid.profile.name = from_uid.profile.name
-        if not to_uid.profile.email:
-            to_uid.profile.email = from_uid.profile.email
-        if not to_uid.profile.gender:
-            to_uid.profile.gender = from_uid.profile.gender
-        if not to_uid.profile.gender_acc:
-            to_uid.profile.gender_acc = from_uid.profile.gender_acc
-        if not to_uid.profile.country:
-            to_uid.profile.country = from_uid.profile.country
+    def _merge_profiles(from_uids, to_uid):
+        """Merge the profiles from `UniqueIdentity` objects"""
+
+        for from_uid in from_uids:
+            if from_uid.profile.is_bot or to_uid.profile.is_bot:
+                to_uid.profile.is_bot = True
+            if not to_uid.profile.name:
+                to_uid.profile.name = from_uid.profile.name
+            if not to_uid.profile.email:
+                to_uid.profile.email = from_uid.profile.email
+            if not to_uid.profile.gender:
+                to_uid.profile.gender = from_uid.profile.gender
+            if not to_uid.profile.gender_acc:
+                to_uid.profile.gender_acc = from_uid.profile.gender_acc
+            if not to_uid.profile.country:
+                to_uid.profile.country = from_uid.profile.country
 
         return to_uid
 
+    def _delete_unique_identities(trxl, uids):
+        """Delete unique identities from the database"""
+
+        for uid in uids:
+            delete_unique_identity_db(trxl, uid)
+
     # Check input values
-    if from_uuid is None:
-        raise InvalidValueError(msg="'from_uuid' cannot be None")
-    if from_uuid == '':
-        raise InvalidValueError(msg="'from_uuid' cannot be an empty string")
+    if from_uuids is None:
+        raise InvalidValueError(msg="'from_uuids' cannot be None")
+    if from_uuids == []:
```

Comment:
It might be better to evaluate it with `len`.

Reply:
The problem I see evaluating this with `len` is that both values `[]` and `''` would return `0` using `len`. That's why I set the check for an empty list here, and another check for an empty string when iterating over the elements of a non-empty list. 
It is also true that when the field `from_uuids` could be pre-validated, so we could assume we could receive an empty list as value but we couldn't receive an empty string, so I'm not sure if this check is redundant or not. Does any of this make sense to you? Thanks in advance!

Reply:
Ok! Makes sense then.